Source/qol/xpbar.cpp

```diff
@@ -0,0 +1,150 @@
+/**
+* @file xpbar.cpp
+*
+* Adds XP bar QoL feature
+*/
+
+#include "common.h"
+#include "control.h"
+#include "DiabloUI/art_draw.h"
+#include "options.h"
+
+#include <array>
+
+namespace devilution {
+
+namespace {
+
+constexpr int BAR_WIDTH = 307;
+constexpr int BAR_HEIGHT = 5;
+
+using ColorGradient = std::array<Uint8, 12>;
+constexpr ColorGradient GOLD_GRADIENT = { 0xCF, 0xCE, 0xCD, 0xCC, 0xCB, 0xCA, 0xC9, 0xC8, 0xC7, 0xC6, 0xC5, 0xC4 };
+constexpr ColorGradient SILVER_GRADIENT = { 0xFE, 0xFD, 0xFC, 0xFB, 0xFA, 0xF9, 0xF8, 0xF7, 0xF6, 0xF5, 0xF4, 0xF3 };
+
+constexpr int BACK_WIDTH = 313;
+constexpr int BACK_HEIGHT = 9;
+
+Art xpbarArt;
+
+void DrawBar(const CelOutputBuffer &out, int x, int y, int width, const ColorGradient &gradient)
+{
+	FastDrawHorizLine(out, x, y + 1, width, gradient[gradient.size() * 3 / 4 - 1]);
+	FastDrawHorizLine(out, x, y + 2, width, gradient[gradient.size() - 1]);
+	FastDrawHorizLine(out, x, y + 3, width, gradient[gradient.size() / 2 - 1]);
+}
+
+void DrawEndCap(const CelOutputBuffer &out, int x, int y, int idx, const ColorGradient &gradient)
+{
+	SetPixel(out, x, y + 1, gradient[idx * 3 / 4]);
+	SetPixel(out, x, y + 2, gradient[idx]);
+	SetPixel(out, x, y + 3, gradient[idx / 2]);
+}
+
+} // namespace
+
+void InitXPBar()
+{
+	if (sgOptions.Gameplay.bExperienceBar) {
+		LoadMaskedArt("data\\xpbar.pcx", &xpbarArt, 1, 1);
```

Comment:
Should throw an error screen if file is not found (same as health bar)

Reply:
Done